rbc/omnisci_backend/numpy_funcs.py

```diff
@@ -272,10 +279,7 @@ def impl(a, initial=None):
 @expose_and_overload(np.mean)
 @extending.overload_method(ArrayPointer, 'mean')
 def omnisci_array_mean(x):
-    if isinstance(x.eltype, types.Integer):
-        zero_value = 0
-    elif isinstance(x.eltype, types.Float):
-        zero_value = np.nan
+    zero_value = np.nan  # 0 if x.dtype is int
```

Comment:
I think `mean` on integer input should return float value, that is, `mean` is a reduction op that promotes integers to float:
```suggestion
    zero_value = np.nan
```